### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by ExceptionsSG** [Orderella/PopupDialog#9]
When dialog is active, CPU activity increased to 45% from 3%. CPU usage will only drop after dialog is dismissed

**Comment by mwfire** [comment]
@ExceptionsSG Thanks, I'll check on that!

**Comment by mwfire** [comment]
@ExceptionsSG Ok, as I suspected this is due to the overlay blur view, which is actually a live blur view. That is, the underlying view is updated in realtime if something changes.

You can test this behavior by switching to landscape mode. However, I agree that it might be better to turn this live feature off by default. A simple dialog might not be worth using that much resources. So it's probably a good idea to opt in on this feature via appearance.

What do you think?

**Comment by leongwq** [comment]
@mwfire I've replaced the live blur with just a simple blur. It still looks good and it isn't using much resource.

```
    var blurEffect = UIBlurEffect(style: UIBlurEffectStyle.Light)
    var blurEffectView = UIVisualEffectView(effect: blurEffect)
    blurEffectView.autoresizingMask = [.FlexibleWidth, .FlexibleHeight]
    addSubview(blurEffectView)
```

**Comment by mwfire** [comment]
@leongwq Yes, I had the native UIVisualEffectView in the initial version. However, it is not really customizable, that's why I switched to FXBlurView. The blur radius is quite high, and it complains when being faded in, as setting alpha values on the effect view is not supported.

Exposing the liveBlur property while setting it to `false` by default might be an alternative. I'll give it a try and let you know!

**Comment by mwfire** [comment]
@ExceptionsSG @leongwq Pushed to the development branch, commit a06cfe1
CPU usage is almost unnoticeable with liveBlur turned off.

Thanks again for bringing that up, good catch!

If Travis is fine with this commit I will push it as version 0.2.2 asap

**Comment by mwfire** [comment]
Released to v0.2.2. Thanks!
